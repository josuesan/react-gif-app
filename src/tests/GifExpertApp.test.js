import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test to Gif App', () => {
  test('should render', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a list opf categories', () => {
    const categories = ['One punch man', 'Naruto'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });


});