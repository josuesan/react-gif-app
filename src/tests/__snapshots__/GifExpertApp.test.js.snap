// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to Gif App should have a list opf categories 1`] = `
<Fragment>
  <h1>
    Gift Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch man"
      key="One punch man"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`Test to Gif App should render 1`] = `
<Fragment>
  <h1>
    Gift Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch man"
      key="One punch man"
    />
  </ol>
</Fragment>
`;
